packages/openapi3/test/tsp-openapi3/output/petstore-sample/main.tsp

```diff
@@ -0,0 +1,107 @@
+import "@typespec/http";
+import "@typespec/openapi";
+import "@typespec/openapi3";
+
+using Http;
+using OpenAPI;
+
+@service({
+  title: "Swagger Petstore",
+})
+@info({
+  version: "1.0.0",
+  license: {
+    name: "MIT",
+  },
+})
+namespace SwaggerPetstore;
+
+model Pet {
+  id: int64;
+  name: string;
+  tag?: string;
+}
+
+model Pets is Pet[];
+
+model Error {
+  code: int32;
+  message: string;
+}
+
+/**
+ * A paged array of pets
+ */
+model listPets200ApplicationJsonResponse {
+  /**
+   * A link to the next page of responses
+   */
+  @header("x-next") xNext?: string;
+
+  @statusCode statusCode: 200;
+  @bodyRoot body: Pets;
+}
+
+/**
+ * unexpected error
+ */
+@defaultResponse
+model listPetsDefaultApplicationJsonResponse {
+  @bodyRoot body: Error;
+}
+
+/**
+ * Null response
+ */
+model createPets201Response {
+  @statusCode statusCode: 201;
+}
+
+/**
+ * unexpected error
+ */
+@defaultResponse
+model createPetsDefaultApplicationJsonResponse {
+  @bodyRoot body: Error;
+}
+
+/**
+ * Expected response to a valid request
+ */
+model showPetById200ApplicationJsonResponse {
+  @statusCode statusCode: 200;
+  @bodyRoot body: Pet;
+}
+
+/**
+ * unexpected error
+ */
+@defaultResponse
+model showPetByIdDefaultApplicationJsonResponse {
+  @bodyRoot body: Error;
+}
+
+@tag("pets")
+@route("/pets")
+@get
+op listPets(
+  /**
+   * How many items to return at one time (max 100)
+   */
+  @query limit?: int32,
+): listPets200ApplicationJsonResponse | listPetsDefaultApplicationJsonResponse;
```

Comment:
Is this really our expectation to make up new model (e.g. `listPets200ApplicationJsonResponse`) which is not defined by users?

Reply:
I mentioned this in the description:
> I can definitely pare this down with some simple heuristics that take into account what default statusCode/contentTypes are, and extract the referenced body type directly in the operation's return signature. I can also eliminate the @get decorators, @route("/") routes, and likely use some of the response models provided by TypeSpec.Http.
> 
> However - if I'm using this tool to convert from OpenAPI3 to TypeSpec - I thought it might be preferable to be more explicit in the generated output so there's no mystery on how things actually get defined. Will be interested in feedback on this.

Originally I actually had responses represented by anonymous model expressions. Unfortunately you can't assign decorators to a model expression - and you use a decorator in TypeSpec to say which response is the `default` one. Without setting this decorator, default responses were showing up under 200 status code which isn't what you want if default is actually meant to be an error.

Reply:
Emmmm, do we expect customers use these generated tsp to generate SDK? If not I'm OK with these models. Further, do we expect customers use these generated tsp to generate swagger? If not I'm OK as well. Otherwise, I think there will be problems for SDK generation.

Reply:
> Otherwise, I think there will be problems for SDK generation.

I'm admittedly still ramping up on the SDK generation side of things with TypeSpec - what specific concerns would there be with having these models generated? The names are admittedly verbose, but I wouldn't expect the SDK emitters to treat this any differently than if an author wrote these themselves.